core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackInstanceStatusChecker.java

```diff
@@ -47,7 +47,7 @@ public List<CloudVmInstanceStatus> queryInstanceStatuses(Stack stack, List<Cloud
             Location location = location(region(stack.getRegion()), availabilityZone(stack.getAvailabilityZone()));
             CloudContext cloudContext = CloudContext.Builder.builder()
                     .withId(stack.getId())
-                    .withName(stack.getName())
+                    .withName(stack.isDetached() ? stack.getDisplayName() : stack.getName())
```

Comment:
I don't like this for 2 reasons, you'll need to know do this in any new locations you're building a CloudContext and when I think of functionality around a DisplayName field, I would not expect it to change any behavior outside of what the customer ends up seeing.